### PR TITLE
Add a table of contents to readme.md

### PR DESCRIPTION
**Pull request by markthethomas**
Thought adding a streamlined TOC might make getting to some of the elements that are further down in the readme a little easier for reference usage. I didn't include every header/section since it ended up being too long :+1:

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -4,6 +4,15 @@
 
 [![Build Status: Linux](https://travis-ci.org/sindresorhus/ava.svg?branch=master)](https://travis-ci.org/sindresorhus/ava) [![Build status: Windows](https://ci.appveyor.com/api/projects/status/igogxrcmhhm085co/branch/master?svg=true)](https://ci.appveyor.com/project/sindresorhus/ava/branch/master) [![Coverage Status](https://coveralls.io/repos/sindresorhus/ava/badge.svg?branch=master&service=github)](https://coveralls.io/github/sindresorhus/ava?branch=master) [![Join the chat](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/sindresorhus/ava?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
+Table of contents
```
Comment (sindresorhus) [inline readme.md]: Use the `##` header style instead of underline.

Comment (sindresorhus) [inline readme.md]: and `Table of Contents`

File: readme.md [hunk 2]
```diff
@@ -4,6 +4,15 @@
...
+=================
+* [Usage](#usage)
```
Comment (sindresorhus) [inline readme.md]: `*` → `-`

File: readme.md [hunk 3]
```diff
@@ -4,6 +4,15 @@
...
+* [CLI Usage](#cli)
+* [Documentation](#documentation)
+* [API](#api)
+* [Assertions](#assertions)
+* [FAQ](#api)
```
Comment (sindresorhus) [inline readme.md]: Can you place the TOC below the below intro?

**Comment by sindresorhus** [comment]
https://github.com/RichardLitt/docs/blob/master/amending-a-commit-guide.md

**Comment by sindresorhus** [comment]
Good idea!

File: readme.md (hunk 3 above)
Comment (markthethomas) [inline readme.md]: Will do! :+1:

**Comment by markthethomas** [comment]
Oo yes, now to learn how to amend my commit :+1:

**Comment by sindresorhus** [comment]
@markthethomas You don't have to amend. You can just push new commits too. I usually squash commits and cleanup afterwards anyways ;)

**Comment by markthethomas** [comment]
Sounds good! Need to learn though — I'm a solo dev where I work and don't get to use Git in these ways all that often (usually just me talking to me lol) :)

**Comment by sindresorhus** [comment]
Open source is definitely the best way to learn to use Git ;)

**Comment by markthethomas** [comment]
Yes indeed! I *think* I just rebased and squashed the PR correctly?

**Comment by markthethomas** [comment]
wait no — left the old TOC in there. Fixing!

**Comment by sindresorhus** [comment]
Btw, noticed that your created the PR from `master`. First rule of PRs is to never do it from master, since that makes it hard to create multiple PRs. Create a new branch for each PR you do. Master is for syncing with upstream.

File: readme.md [hunk 4]
```diff
@@ -6,6 +6,14 @@
 
 Even though JavaScript is single-threaded, IO in Node.js can happen in parallel due to its async nature. AVA takes advantage of this and runs your tests concurrently, which is especially beneficial for IO heavy tests. In addition, test files are run in parallel as separate processes, giving you even better performance and a isolated environment for each test file. [Switching](https://github.com/sindresorhus/pageres/commit/663be15acb3dd2eb0f71b1956ef28c2cd3fdeed0) from Mocha to AVA in Pageres brought the test time down from 31 sec to 11 sec. Having tests run concurrently forces you to write atomic tests, meaning tests that don't depend on global state or the state of other tests, which is a great thing!
 
+## Table of contents
+* [Usage](#usage)
```
Comment (sindresorhus) [inline readme.md]: Use `-` instead of `*` for the items

File: readme.md [hunk 5]
```diff
@@ -550,4 +558,3 @@ Concurrency is not parallelism. It enables parallelism. It's about dealing with,
 	<br>
 	<br>
 </div>
-
```
Comment (sindresorhus) [inline readme.md]: Don't remove this.

**Comment by markthethomas** [comment]
Oh ok — thanks for letting me know! I'll abandon this and create a new branch

**Comment by sindresorhus** [comment]
No, you can finish this one first.

**Comment by markthethomas** [comment]
:+1: (thanks for your patience while I fumble & learn)

**Comment by markthethomas** [comment]
should be good to merge unless there's anything else that needs adding :)

**Comment by sindresorhus** [comment]
I've commented about https://github.com/sindresorhus/ava/pull/168/files#r44228402 twice (`*` → `-`), but I can fix it in the merge.

**Comment by markthethomas** [comment]
Argh -- my bad

Mark

Sent from my iPhone

@markthethomas everywhere

**Comment by sindresorhus** [comment]
No worries :)

**Pull request closed without merging**